### PR TITLE
Fix all the test, remove leftovers from transition

### PR DESCRIPTION
**Pull request by Itxaka**
Remove all the mentions to rancheros-operator in tests, imports and
dependencies.

Restore e2e scripts and test targets on makefile

Fix all lint issues

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+WORKDIR /src
+COPY go.mod go.sum /src/
+RUN go mod download
+COPY main.go /src/
+COPY pkg /src/pkg
+COPY cmd /src/cmd
+# Set arg/env after go mod download, otherwise we invalidate the cached layers due to the commit changing easily
+ARG VERSION=0.0.0
+ENV VERSION=${VERSION}
+RUN go build -ldflags "-extldflags -static -s -X github.com/rancher/elemental-operator/version.Version=$VERSION" -o /usr/sbin/elemental-operator
```
Comment (mudler) [inline Dockerfile:12]:
```suggestion
RUN go build -ldflags "-extldflags -static -w -s -X github.com/rancher/elemental-operator/version.Version=$VERSION" -o /usr/sbin/elemental-operator
```

File: tests/e2e/upgrades_test.go [hunk 2]
```diff
@@ -267,6 +268,7 @@ var _ = Describe("ManagedOSImage e2e tests", func() {
 		})
 
 		It("overwrites default container spec from a channel", func() {
+			Skip("Requires a new image for discovery plugin")
```
Comment (mudler) [inline tests/e2e/upgrades_test.go:271]: is this valid still? or can't we just release a fixed one before going ahead?

File: Dockerfile (hunk 1 above)
Comment (Itxaka) [reply Dockerfile:12]: :+1: will add it!

**Comment by davidcassany** [comment]
@Itxaka note some of the former `ros-installer` tests are failing (60min timeout) because they are executed in parallel and they overlap. Some of them open a server at a static local URL so these should not be executed in parallel as the server will misbehave then. I'd suggest to tests everything in sequence and later on improve the unit tests.

**Comment by Itxaka** [comment]
ah nice, thanks for the heads up @davidcassany I will mark those as serial!

**Comment by Itxaka** [comment]
Some extra lint issues that come due to newer updates to main, will tackle those afterwards

**Review by mudler — APPROVED** [review]
Looks good here! Let's go indeed ahead with it and tackle linting separately

**Pull request merged**